src/SelfServe/Example/Example.tsx

```diff
@@ -0,0 +1,189 @@
+  @CustomElement()
+    - input: JSX.Element | (currentValues: Map<string, InputType> => Promise<JSX.Element>)
+    - role: Display a custom element by either passing the element itself, or by passing a function that takes the current values
+            and renders a Component / JSX.Element.
+
+            In this example, we first use a static JSX.Element to show a description text. We also declare a CustomComponent, that 
+            takes a Map of propertyName -> value, as input. It uses this to display a Hoverable Card which shows a snapshot of 
+            the current values.
+  */
+  @CustomElement(descriptionElement)
+  static description: string;
+
+  /*
+  @ParentOf()
+    - input: string[]
+    - role: Determines which UI elements are the children of which UI element. An array containing the names of the child properties 
+            is passsed. You need to make sure these children are declared in this Class as proeprties.
+  */
+  @ParentOf(["choiceInput", "booleanInput", "stringInput", "numberSliderInput", "numberSpinnerInput"])
+  @CustomElement(renderText("Hover to see current values..."))
+  static currentValues: string;
+
+  /*
+  @Label()
+    - input: string | () => Promise<string>
+    - role: Adds a label for the UI element. This is ignored for a custom element but is required for all other properties.
+  */
+  @Label("Choice")
+
+  /*
+  @PropertyInfo()
+    - input: Info | () => Promise<Info>
+    - role: Display an Info bar above the UI element for this property.
+  */
+  @PropertyInfo(choiceInfo)
+
+  /*
+  @ChoiceInput()
+    - input: ChoiceItem[] | () => Promise<ChoiceItem[]>
+    - role: Display a dropdown with choices.
+  */
+  @ChoiceInput(choiceOptions)
+  static choiceInput: ChoiceItem;
+
+  @Label("Boolean")
+  /*
+  @BooleanInput()
+    - input: 
+        trueLabel : string | () => Promise<string>
+        falseLabel : string | () => Promise<string>
+    - role: Add a boolean input eith radio buttons for true and false values.
+  */
+  @BooleanInput({
+    trueLabel: "allowed",
+    falseLabel: "not allowed"
+  })
+  static booleanInput: boolean;
+
+  @Label("String")
+  /*
+  @PlaceHolder()
+    - input: string | () => Promise<string>
+    - role: Adds a placeholder for the string input
+  */
+  @Placeholder("instance name")
+  static stringInput: string;
+
+  @Label("Slider")
+
+  /*
+  @OnChange()
+    - input: (currentValues: Map<string, InputType>, newValue: InputType) => Map<string, InputType>
+    - role: Takes a Map of current values and the newValue for this property as inputs. This is called when a property 
+            changes its value in the UI. This can be used to change other input values based on some other input.
+            
+            The new Map of propertyName -> value is returned.
+
+            In this example, the onSliderChange function sets the spinner input to the same value as the slider input
+            when the slider in moved in the UI.
+  */
+  @OnChange(onSliderChange)
```

Comment:
This is to modify another UI element based on change in value of this UI element. 
It is an interesting use case to make it directly dependent declaritively in the Example.tsx file itself instead of having to depend on the user processing and setting values in a map.
Will keep it like this for this initial merge, will revisit it in the next PR.